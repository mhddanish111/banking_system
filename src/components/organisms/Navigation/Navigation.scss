@import '../../../styles/colors.scss';
@import '../../../styles/mixins.scss';

.nav-container {
	height: calc(100vh - 70px);
	width: 270px;
	position: relative;
	overflow-y: scroll;
	overflow-x: hidden;
	transition: all 0.5s ease-in-out;
	@include tablet-and-under {
		width: 100vw;
		position: absolute;
		transition: all 0.6s ease-in-out;
		top: 0;
		left: -100vw;
	}
	&::-webkit-scrollbar {
		display: none;
	}
}
.nav {
	min-height: 91vh;
	width: 250px;
	background-color: $background-color_8;
	position: absolute;
	top: 0px;
	left: 00;
	box-shadow: 1px 1px 10px rgba(198, 189, 248, 0.825);
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	overflow: hidden;
	padding: 30px 0 20px 10px;
	@include tablet-and-under {
		// width: 100%;
		position: absolute;
	}
}
.nav-close {
	@include tablet-and-under {
		left: 0px;
		width: 100%;
		z-index: 1000;
	}
}
.nav-option {
	width: 250px;
	height: 60px;
	display: flex;
	align-items: center;
	padding: 0 30px 0 20px;
	gap: 20px;
	transition: all 0.1s ease-in-out;
	&:hover {
		border-left: 5px solid #a2a2a2;
		background-color: $background-color_9;
		cursor: pointer;
	}
	h3 {
		font-size: 20px;
	}
	a {
		text-decoration: none;
		cursor: pointer;
		color: $color-black;
	}
	@include mobile {
		gap: 25px;
		h3 {
			font-size: 12px;
		}
	}
}
.nav-img {
	height: 30px;
	@include mobile {
		height: 25px;
	}
}
.nav-upper-options {
	display: flex;
	flex-direction: column;
	align-items: center;
	gap: 30px;
	@include mobile {
		gap: 15px;
	}
}
.option1 {
	border-left: 5px solid #010058af;
	background-color: $background-color_10;
	color: $color-black;
	cursor: pointer;
	&:hover {
		border-left: 5px solid #010058af;
		background-color: $background-color_10;
	}
}