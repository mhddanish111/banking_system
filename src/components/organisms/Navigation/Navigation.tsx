import React from 'react';
import { Link, useNavigate } from 'react-router-dom';
import GetContext from '../../../context/GetContext';
import { deleteCookie } from '../../../utils';
import { resetContext } from '../../../context/actions';
import './Navigation.scss';

const Navigation: React.FC = React.memo(() => {
  const { dispatch }: any = GetContext();
  const navigate = useNavigate();
  const handleLogOut = (e: any) => {
    e.preventDefault();
    deleteCookie();
    dispatch(resetContext());
    navigate('/');
  };
  return (
    <div className="nav-container">
      <nav className="nav">
        <div className="nav-upper-options">
          <div className="nav-option option1">
            <Link to="/dashboard">
              <h3>Dashboard</h3>
            </Link>
          </div>

          <div className="nav-option">
            <Link to="/account">
              <h3>Account</h3>
            </Link>
          </div>

          <div className="nav-option">
            <Link to="/credit-card">
              <h3>Credit Card</h3>
            </Link>
          </div>

          <div className="nav-option">
            <Link to="/create-account">
              <h3>Create Account</h3>
            </Link>
          </div>

          <div className="nav-option">
            <Link to="/profile">
              <h3> Profile</h3>
            </Link>
          </div>

          <div className="nav-option logout">
            <a href="/#" onClick={(e) => handleLogOut(e)}>
              <h3>Logout</h3>
            </a>
          </div>
        </div>
      </nav>
    </div>
  );
});

export default Navigation;
