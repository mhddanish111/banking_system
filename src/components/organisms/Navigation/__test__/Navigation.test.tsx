import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import Navigation from '../Navigation';

describe('<Navigation />', () => {
  const mockedUsedNavigate = jest.fn();
  jest.mock('react-router-dom', () => ({
    ...(jest.requireActual('react-router-dom') as any),
    useNavigate: () => mockedUsedNavigate,
  }));
  const wrapped = shallow(
    <BrowserRouter>
      <Navigation />
    </BrowserRouter>,
  );
  it('should render the Title Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
