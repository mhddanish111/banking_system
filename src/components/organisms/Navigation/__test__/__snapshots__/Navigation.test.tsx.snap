// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render the Title Component correctly 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "createHref": [Function],
      "encodeLocation": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Navigation />
</Router>
`;
